sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +767,80 @@
             return monoError(logger, ex);
         }
     }
+
+    private static final Duration DEFAULT_POLL_INTERVAL = Duration.ofSeconds(5);
+
+    /**
+     * Long Running Operation for the purchaseSearch.
+     *
+     * @param searchId ID of the search
+     * @param lroDuration Timelapse of the poll request.
+     * @return A {@link Mono} containing a {@link Response} for the operation
+     */
+
+    public PollerFlux<Void, PhoneNumberSearch> beginPurchaseSearch(String searchId, Duration lroDuration) {
+
+        final Duration duration = lroDuration != null ? lroDuration : DEFAULT_POLL_INTERVAL;
+        return new PollerFlux<Void, PhoneNumberSearch>(duration,
+            activationOperation(searchId),
+            createPollOperation(searchId),
+            (activationResponse, pollingContext) -> Mono.error(new RuntimeException("Cancellation is not supported")),
+            fetchResultOperation(searchId));
+    }
+
+    private Function<PollingContext<Void>, Mono<Void>> activationOperation(String searchId) {
+
+        return (pollingContext) -> withContext(context -> purchaseSearchWithResponse(searchId, context)
+            .flatMap(response -> Mono.just(response.getValue())));
+    }
+
+    private Function<PollingContext<Void>, Mono<PollResponse<Void>>> createPollOperation(String searchId) {
+        return (pollingContext) -> {
+            try {
+                return withContext(context -> getSearchByIdWithResponse(searchId, context))
+                    .flatMap(this::getSearchByIdStatus);
```

Comment:
`getSearchByIdStatus` isn't clear as a varable, it should be something more like `getSearchByIdResponse`